components/raftstore/src/store/util.rs

```diff
@@ -911,16 +913,22 @@ struct RegionReadProgressCore {
     applied_index: u64,
     // TODO: after region merged, the region's key range is extended and this
     // region wide `safe_ts` should reset to `min(source_safe_ts, target_safe_ts)`
-    safe_ts: u64,
+    read_state: ReadState,
     // `pending_items` is a *sorted* list of `(apply index, safe ts)` item
```

Comment:
Do we need to update the corresponding comments here?

Reply:
I think it is okey, `ReadState` just a wrapper of `(apply_index, safe_ts)`